provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,445 @@
+  @GetMapping(params = {"address-state"})
+  public Location.Bundle searchByState(
+      @RequestParam("address-state") String state,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("address-state", state)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  /** Search by family & given name. */
+  @GetMapping(params = {"address-postalcode"})
+  public Location.Bundle searchByZip(
+      @RequestParam("address-postalcode") String zip,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("address-postalcode", zip)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  private List<LocationWrapper> searchIdentifier(MultiValueMap<String, String> parameters) {
+    LocationWrapper.LocationWrapperBuilder locationWrapper =
+        LocationWrapper.builder()
+            .careSitesResponse(CareSitesResponse.builder().build())
+            .providerResponse(ProviderResponse.builder().build())
+            .providerServicesResponse(ProviderServicesResponse.builder().build());
+    String identifier = parameters.getFirst("identifier");
+    ProviderServicesResponse providerServicesResponse = ppmsClient.providerServicesById(identifier);
+    locationWrapper.providerResponse(
+        ProviderResponse.builder()
+            .value(
+                singletonList(
+                    ProviderResponse.Value.builder()
+                        .providerIdentifier(
+                            Integer.parseInt(parameters.get("identifier").toArray()[0].toString()))
+                        .build()))
+            .build());
+    if (providerServicesResponse.value().isEmpty()) {
+      locationWrapper.providerResponse(ppmsClient.providersForId(identifier));
+      locationWrapper.careSitesResponse(ppmsClient.careSitesById(identifier));
+    }
+    return singletonList(
+        locationWrapper.providerServicesResponse(providerServicesResponse).build());
+  }
+
+  private List<LocationWrapper> searchName(MultiValueMap<String, String> parameters) {
+    LocationWrapper.LocationWrapperBuilder locationWrapper =
+        LocationWrapper.builder()
+            .careSitesResponse(CareSitesResponse.builder().build())
+            .providerResponse(ProviderResponse.builder().build())
+            .providerServicesResponse(ProviderServicesResponse.builder().build());
+    List<LocationWrapper> filteredResults = new ArrayList<>();
+    totalRecords = 1;
+    String name = parameters.get("name").toArray()[0].toString();
+    locationWrapper.providerResponse(ppmsClient.providersForName(trimIllegalCharacters(name)));
```

Comment:
This `locationWrapper` isn't included in the results you return... it appears to be used just as a container for the results from the `providersForName` call.